### PR TITLE
test: Check if we end up in the expected volume in Gen1 `Navigator`

### PR DESCRIPTION
**Pull request by andiwand**
I think this is one of the big navigation pitfalls. If we end up in the wrong volume the navigation is practically lost. This is too expensive to check in production so I opted for an `assert`.

Not sure if that is a great solution since users might try to measure performance with non release build by accident. But that was the only way I could think of.

blocked by
- https://github.com/acts-project/acts/pull/3521
- https://github.com/acts-project/acts/pull/3481

### CODE REVIEW

**Comment by paulgessinger** [comment]
One alternative to the `assert` would be an explicit configuration that sets up a *hardened* navigation mode where this check is performed.

**Comment by andiwand** [comment]
Problem with the GSF and GX2F can be observed here @benjaminhuth @AJPfleger

**Comment by andiwand** [comment]
> I really like the idea to have this (conditionally) be a hard failure.
> 
> One alternative to the `assert` would be an explicit configuration that sets up a _hardened_ navigation mode where this check is performed.
> 
> On another note: we force-enable `assert` even on non-Debug builds in the CI so that we gain exposure to assertion failures. So, depending on the slowdown, this might be an issue.

I did not really think of the CI. We have to see how bad it is. But I would trade some performance for this because otherwise we might just not end up checking this at all

**Comment by andiwand** [comment]
physmon ttbar changes because we now cut on different particles

**Comment by AJPfleger** [comment]
> Problem with the GSF and GX2F can be observed here @benjaminhuth @AJPfleger

I might have fixed the gx2f issue with
- https://github.com/acts-project/acts/pull/3411

ready to update? :)

**Comment by andiwand** [comment]
@AJPfleger it is not failing inside the unit test

```
13:43:23    Navigator      VERBOSE   No Volume | Initialization.
13:43:23    Navigator      VERBOSE   No Volume | Slow start initialization through search.
13:43:23    Navigator      VERBOSE   No Volume | Starting from position (750.0221, -749.9779, -0.1743) and direction (-0.2816, 0.9595, 0.0001)
13:43:23    Navigator      VERBOSE   TestVolume | Start volume resolved.
ActsUnitTestGx2f: /__w/acts/acts/Core/include/Acts/Propagator/Navigator.hpp:325: void Acts::Navigator::initialize(propagator_state_t&, const stepper_t&) const [with propagator_state_t = Acts::PropagatorState<Acts::PropagatorOptions<Acts::EigenStepper<>::Options, Options, Acts::ActionList<Acts::Experimental::Gx2Fitter<const Acts::Propagator<Acts::EigenStepper<>, Acts::Navigator>, Acts::VectorMultiTrajectory>::Actor<Acts::GenericBoundTrackParameters<Acts::ParticleHypothesis> > >, Acts::AbortList<Acts::Experimental::Gx2Fitter<const Acts::Propagator<Acts::EigenStepper<>, Acts::Navigator>, Acts::VectorMultiTrajectory>::Aborter<Acts::GenericBoundTrackParameters<Acts::ParticleHypothesis> >, Acts::PathLimitReached> >, Acts::EigenStepper<>::State, State, Acts::Experimental::Gx2FitterResult<Acts::VectorMultiTrajectory> >; stepper_t = Acts::EigenStepper<>]: Assertion `state.navigation.startVolume->inside( stepper.position(state.stepping), state.options.surfaceTolerance) && "We did not end up inside the volume."' failed.
unknown location(0): fatal error: in "Gx2fTest/UpdatePushedToNewVolume": signal: SIGABRT (application abort requested)
```

**Comment by andiwand** [comment]
This is not blocked by https://github.com/acts-project/acts/pull/3481. `lowestTrackingVolume` determines a volume for a position we are not in which then causes the assert to fire.

File: CI/physmon/workflows/physmon_trackfinding_ttbar_pu200.py
Comment (AJPfleger) [inline CI/physmon/workflows/physmon_trackfinding_ttbar_pu200.py:1]: Why do we add the cuts? I saw the commit `fix ttbar physmon by cutting for particles in beampipe` but it is not really clear for me. Also `eta`, `pT`, and `removeNeutral` seem not related to the beampipe?

**Review by AJPfleger — COMMENTED** [review]
Looks good in general :)

File: CI/physmon/workflows/physmon_trackfinding_ttbar_pu200.py
Comment (andiwand) [reply CI/physmon/workflows/physmon_trackfinding_ttbar_pu200.py:1]: true those should not be necessary for that cause. I will try to revert them

File: CI/physmon/workflows/physmon_trackfinding_ttbar_pu200.py [hunk 1]
```diff
@@ -63,6 +64,13 @@
         setup.trackingGeometry,
         setup.field,
         rnd=rnd,
+        preSelectParticles=ParticleSelectorConfig(
+            rho=(0.0, 24 * u.mm),
+            absZ=(0.0, 1.0 * u.m),
+            eta=(-3.0, 3.0),
+            pt=(150 * u.MeV, None),
+            removeNeutral=True,
```
Comment (andiwand) [inline CI/physmon/workflows/physmon_trackfinding_ttbar_pu200.py:72]:
```suggestion
```

**Review by AJPfleger — APPROVED** [review]

**Comment by acts-project-service** [comment]
## :red_circle: Athena integration test results [1f8110466432583d53d981d4f42ee17a03759da3]
### :red_circle: Some tests have failed!
Please investigate the [pipeline](https://gitlab.cern.ch/acts/acts-athena-ci/-/pipelines/8039563)!

|status|job|report|
|------|---|------|
|:green_circle: | [report_pull_request](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996362)||
|:green_circle: | [run_art_test: test_data18_13TeV_1000evt](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996361) | [:paperclip:](https://acts-athena-outputs.web.cern.ch/test_data18_13TeV_1000evt/pulls/3442/)|
|:green_circle: | [run_art_test: test_ttbarPU40_reco](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996360) | [:paperclip:](https://acts-athena-outputs.web.cern.ch/test_ttbarPU40_reco/pulls/3442/)|
|:green_circle: | [test_ActsDumpGeometryIdentifiers](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996359)||
|:red_circle: | [test_ActsCheckObjectCountsCachedWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996358)||
|:red_circle: | [test_ActsCheckObjectCountsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996357)||
|:green_circle: | [test_ActsEFTrackFit](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996356)||
|:green_circle: | [test_ActsPersistifySeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996355)||
|:green_circle: | [test_ActsBenchmarkWithSpot](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996354)||
|:green_circle: | [test_ActsAnalogueClustering](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996353)||
|:green_circle: | [test_ActsWorkflowHeavyIons](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996352)||
|:green_circle: | [test_ActsWorkflowFastTracking](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996351)||
|:green_circle: | [test_ActsWorkflowCached](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996350)||
|:green_circle: | [test_ActsWorkflow](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996349)||
|:green_circle: | [test_ActsValidateAmbiguityResolution](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996348)||
|:green_circle: | [test_ActsValidateResolvedTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996347)||
|:green_circle: | [test_ActsValidateTracks](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996346)||
|:green_circle: | [test_ActsValidateActsCoreSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996345)||
|:green_circle: | [test_ActsValidateActsSpacePoints](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996344)||
|:green_circle: | [test_ActsValidateSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996343)||
|:green_circle: | [test_ActsValidateOrthogonalSeeds](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996342)||
|:green_circle: | [test_ActsValidateClusters](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996341)||
|:green_circle: | [test_ActsPersistifyEDM](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996340)||
|:green_circle: | [test_ActsGSFRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996339)||
|:green_circle: | [test_ActsKfRefitting](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996338)||
|:green_circle: | [test_ActsExtrapolationAlgTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996337)||
|:green_circle: | [test_ActsITkTest](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996336)||
|:green_circle: | [run_workflow_tests_run4_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996335)||
|:green_circle: | [run_workflow_tests_run2_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996334)||
|:green_circle: | [run_workflow_tests_run2_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996333)||
|:green_circle: | [run_workflow_tests_run3_mc](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996332)||
|:green_circle: | [run_workflow_tests_run3_data](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996331)||
|:green_circle: | [run_unit_tests](https://gitlab.cern.ch/acts/acts-athena-ci/-/jobs/42996330)||
